app/shared/app-data/src/androidMain/kotlin/domain/torrent/client/RemoteCall.kt

```diff
@@ -72,4 +96,40 @@ class RetryRemoteCall<I : IInterface>(
             null
         }
     }
+}
+
+/**
+ * Wrapper for call which takes a continuation-like argument and returns [IDisposableHandle],
+ * which means this is a asynchronous RPC call.
+ *
+ * [IDisposableHandle] takes responsibility to pass cancellation to server.
+ */
+suspend inline fun <I : IInterface, T> RemoteCall<I>.callSuspendCancellable(
+    crossinline transact: I.(
+        resolve: (T?) -> Unit,
+        reject: (RemoteContinuationException?) -> Unit
+    ) -> IDisposableHandle?,
+): T = suspendCancellableCoroutine { cont ->
+    val disposable = call {
+        transact(
+            { value ->
+                if (value == null) {
+                    cont.resumeWithException(CancellationException("Remote resume a null value."))
+                } else {
+                    cont.resume(value)
+                }
+            },
+            { exception ->
+                cont.resumeWithException(
+                    exception?.smartCast() ?: Exception("Remote resume a null exception."),
```

Comment:
```suggestion
                    exception?.smartCast() ?: IllegalStateException("Remote resume a null exception."),
```

那是不是实际上不应该允许 exception 为 null (类型上)

Reply:
可能是 null，rpc 如果失败了会 null